File: src/App/src/app/model/menuitem.ts
import { Topping } from "./topping";

export class Menuitem {

  constructor(
      public menuItem_title: string,
      public description: string,
      public price: number,
      public image_URL: string,
      public create_date: Date,
      public last_update: Date,
      public _links: {
        self: {
          href: string
        }
      },
      public toppings?: Topping[],
      public id?: number
  ) {
  }

  public getTitle(){
    return this.menuItem_title;
  }

}

File: src/main/java/com/winsley/sandwich/entities/MenuItem.java
package com.winsley.sandwich.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name="menu_items")
@Getter
@Setter
public class MenuItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "menuItem_id")
    private Long id;

    @Column(name = "create_date")
    @CreationTimestamp
    private Date create_date;

    @Column(name = "description", length = 2500)
    private String description;

    @Column(name = "image_url", length = 2500)
    private String image_URL;

    @Column(name = "last_update")
    @UpdateTimestamp
    private Date last_update;

    @Column(name = "price")
    private BigDecimal price;

    @Column(name = "menuItem_title")
    private String menuItem_title;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "menuItem")
    private Set<CartItem> cartItems = new HashSet<>();

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "menuItem_id")
    private Set<Topping> toppings = new HashSet<>();
}
